### PR TITLE
KOGITO-1234 implements redesign of timeline (wip)

### PR DESCRIPTION
**Pull request by srambach**
This implements a redesign of the process timeline on the details page. I need help with a few of the coding things:
- It needs to have the kebab dropdown menu on the individual timeline items built. 
- It also needs to have the correct state icon displayed. 
There are TODOs in the code marking these items. The easiest way might be for someone to do a PR implementing those pieces to this branch and then this PR can be merged after that.

Screenshot:
image

@cristianonicolai @kywalker-rh @lclay2

### CODE REVIEW

**Comment by lclay2** [comment]
@srambach , I took a quick look and it looks good to me. 
- Just one nit/question, in the mockup all the nodes have time details under them except for "Confirm Travel" is that just a demo boo boo? 
- I think you mentioned/suggested adding tooltips, I think that would be nice for the status and task type icons in case the meaning isn't obvious to the user.

**Comment by cristianonicolai** [comment]
@srambach looks quite nice!

**Comment by srambach** [comment]
> * Just one nit/question, in the mockup all the nodes have time details under them except for "Confirm Travel" is that just a demo boo boo?

@lclay2 I think that is because that node is actually active and will have the Active icon showing instead of the Completed once that is implemented.

**Comment by lclay2** [comment]
@srambach  Ah, okay I wasn't sure what the time described. So it's basically the last update, if so instead of "Active" for the active one, could it instead provide a time of the last push or refresh interval (1 secs ago) or something like that? I'd suggest adding a "Last updated" prefix, but I think that would be a bit redundant. ;)

File: packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx [hunk 1]
```diff
@@ -18,31 +52,98 @@ const ProcessDetailsTimeline: React.FC<IOwnProps> = ({ loading, data }) => {
...
+                          />
+                        </Tooltip>
+                        {/* <Tooltip content={'Error'}>
+                            <ErrorCircleOIcon
+                              color="var(--pf-global--danger-color--100)"
+                              className="kogito-management-console--timeline-status"
+                            />
+                          </Tooltip> */}
+                      </>
+                    }
```
Comment (AjayJagan) [inline packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx]: we can add this :
 const stateIconCreator = state => {
    switch (state) {
      case ProcessInstanceState.Active:
        return (
          <>
            <OnRunningIcon className="pf-u-mr-sm" />
            Active
          </>
        );
      case ProcessInstanceState.Completed:
        return (
          <>
            <CheckCircleIcon
              className="pf-u-mr-sm"
              color="var(--pf-global--success-color--100)"
            />
            Completed
          </>
        );
      case ProcessInstanceState.Aborted:
        return (
          <>
            <BanIcon className="pf-u-mr-sm" />
            Aborted
          </>
        );
      case ProcessInstanceState.Suspended:
        return (
          <>
            <PausedIcon className="pf-u-mr-sm" />
            Suspended
          </>
        );
      case ProcessInstanceState.Error:
        return (
          <>
            <ErrorCircleOIcon
              className="pf-u-mr-sm"
              color="var(--pf-global--danger-color--100)"
            />
            Error
          </>
        );
    }
  };

**Comment by cristianonicolai** [comment]
@srambach can this be closed or do we still need to apply some of the changes?

**Comment by srambach** [comment]
Implemented in KOGITO-1614

**Pull request closed without merging**